### DISCUSSION TITLE
GitHub Skills: nothing added to commit but untracked files present error

### DISCUSSION THREAD

**Discussion by schalkneethling** [community/community#35476] [Actions] [answered]
Hey All,

I hope this is the appropriate category. I am building a short course using GitHub skills, and specifically, the `skills/template-template` as the basis. I did not change the step 0 workflow but, when I create a new repo from mine(using "Use this template"), the workflow runs, but fails with the following:

```bash
The previous cherry-pick is now empty, possibly due to conflict resolution.
On branch main
If you wish to commit it anyway, use:

    git commit --allow-empty

Otherwise, please use 'git cherry-pick --skip'
Your branch is up to date with 'origin/main'.

You are currently cherry-picking commit fc27034.
  (all conflicts fixed: run "git cherry-pick --continue")
  (use "git cherry-pick --skip" to skip this patch)
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	README.md.tmp
``` 

I am not sure why this happens. If I complete the task that is set-up as step 1, the workflow fails with the same error but, the `README` is actually updated i.e. the appropriate step is expanded. If you want to look at the repo, it can be found here:
https://github.com/The-Mycelium-Network/getting-started-with-github

Here is an example of a failing workflow:
https://github.com/schalkneethling/test-repo/actions/runs/3207539487/jobs/5242521867

Thanks!

**Comment by airtower-luna** [comment, marked as answer]
I suspect your problem is here: https://github.com/schalkneethling/test-repo/actions/runs/3207539487/workflow#L49

Look at what `skills/action-update-step` does if you set `branch_name`: https://github.com/schalkneethling/test-repo/actions/runs/3207539487/workflow#L49

It tries to cherry-pick the latest commit on `main` (which it created just before that) to that branch. Which naturally results in an empty change if `branch_name` is `main`, because the change has already been made.

**Reply by schalkneethling** [reply]
Ah ok, so the docs in the yml file states:

```yml
branch_name:
    description: "If the learner is working in a branch, add the branch name."
```

So, as the learner has just created the repo and has not branched yet, I guess I should _bot_ set a branch name until they have actually branched as part of the course.

https://github.com/skills/action-update-step/blob/main/action.yml#L13

**Reply by schalkneethling** [reply]
If that is correct, I wonder whether the line on [line 51 should remove](https://github.com/skills/template-template/blob/main/.github/workflows/0-start.yml#L51) in step 0 of template/template? As this is the step that is run after the learner created their repo from the template repository. As mentioned here: https://github.com/skills/template-template/blob/main/.github/workflows/0-start.yml#L3

**Reply by schalkneethling** [reply]
Also, maybe this line, "If the learner is working in a branch, add the branch name." should say "If the learner is working in a feature branch, add the branch name."

**Comment by ossanna16** [comment]
Thank you for asking a great question @schalkneethling! Glad you found our community 🙂
